psp/ml/metrics.py

```diff
@@ -0,0 +1,25 @@
+from typing import Callable
+
+import numpy as np
+
+from psp.ml.types import Y
+
+# Base type of a metric.
+# The output is an array because we predict for different timestamps in the future.
+Metric = Callable[[Y, Y], np.ndarray]
+
+
+def mean_absolute_error(y_true: Y, y_pred: Y) -> np.ndarray:
+    return abs(y_true.powers - y_pred.powers)
+
+
+class MeanRelativeError:
+    def __init__(self, cap: float | None = None):
+        self._cap = cap
+
+    def __call__(self, y_true: Y, y_pred: Y) -> np.ndarray:
+        err = abs(y_true.powers - y_pred.powers) / y_true.powers
```

Comment:
becasue of small numbers, do you want to consdier abs(y_true.powers - y_pred.powers) / (0.5*y_true.powers + 0.5*y_pred.powers)

Reply:
Good idea. I find that metric harder to interpret though! The `cap` option fixes the same issue I think? In general I'm not sure how much relative error is going to be useful anyway. MAE might be the way to go!